src/lib/gssapi/krb5/acquire_cred.c

```diff
@@ -595,9 +595,16 @@ kg_cred_set_initial_refresh(krb5_context context, krb5_gss_cred_id_rec *cred,
     set_refresh_time(context, cred->ccache, refresh);
 }
 
+struct _init_verify_creds {
+    krb5_principal srvprinc;
+    krb5_keytab keytab;
+} kg_verify_creds;
```

Comment:
My inclination is that it would be better to just pass two arguments, but maybe I'm missing a complication there.

Reply:
Hi @greghudson ,
I did not want to bloat the APIs by passing multiple separate arguments.

Also when I started working on this I was not 100% sure the two members of the structure were sufficient, so the structure allowed me flexibility to add, remove, change things w/o changing all the intermediate functions that simply pass along this argument without touching it.

Finally the struct allows me the simple option to pass a NULL pointer as indication that verification is not needed. Without it I will have to pass a third boolean option through all these APIs, because it is legal to pass NULL for both the srvprinc and keytab arguments and still want verification, ie when no "keytab" key is present in the store and "verify" is the empty string.

Because of this I would prefer to keep this internal struct, but I am also ok to convert it to three additional argument to pass along if you request it.

Reply:
Understood.

I made some changes (keeping the structure); do they seem okay?

Reply:
Yeah all changes make a lot of sense.
Thanks!